controllers/volumereplication_controller.go

```diff
@@ -64,3 +86,34 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 		For(&replicationv1alpha1.VolumeReplication{}).
 		Complete(r)
 }
+
+func (r VolumeReplicationReconciler) getVolumeHandle(instance *replicationv1alpha1.VolumeReplication) (string, error) {
```

Comment:
```suggestion
func (r VolumeReplicationReconciler) getVolumeHandle(req types.NamespacedName{}) (string, error) {
```
If you do this, you won't need to get the entire `VolumeReplication` object to get volume handle.